Sources/_OpenAPIGeneratorCore/Parser/YamsParser.swift

```diff
@@ -58,9 +65,63 @@ struct YamsParser: ParserProtocol {
             throw OpenAPIVersionError(versionString: "openapi: \(openAPIVersion)")
         }
 
-        return try decoder.decode(
-            OpenAPI.Document.self,
-            from: input.contents
-        )
+        do {
+            return try decoder.decode(
+                OpenAPI.Document.self,
+                from: input.contents
+            )
+        } catch DecodingError.dataCorrupted(let errorContext) {
+            try possibly_throw_parsing_error(context: errorContext, input: input, diagnostics: diagnostics)
+            throw DecodingError.dataCorrupted(errorContext)
+        }
+    }
+
+    /// Detect specific YAML parsing errors to emit nicely formatted errors for IDEs
+    private func possibly_throw_parsing_error(
```

Comment:
```suggestion
    private func checkParsingError(
```

A naming suggestion - let's use camelCase, and suggested slightly different phrase.

Reply:
Done